vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/registration/batch/SampleSpreadsheetLayout.java

```diff
@@ -116,14 +121,27 @@ private void resetChildValues() {
     }
 
     private boolean isInputValid() {
-      binders.forEach(Binder::validate);
-      return binders.stream().allMatch(Binder::isValid);
+      Result<List<NGSRowDTO>, SpreadsheetValidationException> content = sampleRegistrationSpreadsheet.getFilledRows();
+      if(content.isError()) {
+        InformationMessage infoMessage = new InformationMessage(
+            "Incomplete or erroneous metadata found",
+            content.getError().getMessage() + " Row: "+content.getError().getInvalidRow());
+        StyledNotification notification = new StyledNotification(infoMessage);
+        // we need to reload the sheet as the notification popup and removal destroys the spreadsheet UI for some reason...
+        notification.addAttachListener(
+            (ComponentEventListener<AttachEvent>) attachEvent -> sampleRegistrationSpreadsheet.reload());
+        notification.addDetachListener(
+            (ComponentEventListener<DetachEvent>) detachEvent -> sampleRegistrationSpreadsheet.reload());
+        notification.open();
+        return false;
+      }
+      return true;
     }
 
     private List<SampleRegistrationContent> getContent() {
-      List<NGSRowDTO> filledRows = sampleRegistrationSpreadsheet.getFilledRows();
+      Result<List<NGSRowDTO>, SpreadsheetValidationException> filledRows = sampleRegistrationSpreadsheet.getFilledRows();
       List<SampleRegistrationContent> samplesToRegister = new ArrayList<>();
-      filledRows.forEach(row -> {
+      filledRows.getValue().forEach(row -> {
```

Comment:
There is too much happening in the forEach. Please split it. Consider using `map`

Reply:
This code is not part of my PR and I fear I would make it worse using map. :)